lib/merch_calendar/stitch_fix_fiscal_year_calendar.rb

```diff
@@ -87,33 +135,138 @@ def end_of_month(year, merch_month)
     end
 
     # Returns the date that corresponds to the first day in the merch week
+    #
+    # @param [Fixnum] year - the fiscal year
+    # @param [Fixnum] merch_month - the nth month of the fiscal calendar
+    # @param [Fixnum] merch_week - the nth week of the merch month
+    # @return [Date] the start date of the merch week
     def start_of_week(year, month, merch_week)
       start_of_month(year, month) + ((merch_week - 1) * 7)
     end
 
     # Returns the date that corresponds to the last day in the merch week
+    #
+    # @param [Fixnum] year - the fiscal year
+    # @param [Fixnum] merch_month - the nth month of the fiscal calendar
+    # @param [Fixnum] merch_week - the nth week of the merch month
+    # @return [Date] the end date of the merch week
     def end_of_week(year, month, merch_week)
       start_of_month(year, month) + (6 + ((merch_week - 1) * 7))
     end
 
     # Returns the number of weeks in the fiscal year
+    #
+    # @param [Fixnum] year - the fiscal year
+    # @return [Fixnum] the number of weeks within the fiscal year
     def weeks_in_year(year)
       ((start_of_year(year + 1) - start_of_year(year)) / 7).to_i
     end
+
+    # Given any julian date it will return what Fiscal Year it belongs to
+    #
+    # @param [Date] the julian date to convert to its Fiscal Year
+    # @return [Fixnum] the fiscal year that the julian date falls into
+    def merch_year_from_date(date)
+      if end_of_year(date.year) >= date
+        return date.year
+      else
+        return date.year + 1
+      end
+    end
+
+    # Converts a merch month to the correct julian month
+    #
+    # @param merch_month [Fixnum] the merch month to convert
+    # @return [Fixnum] the julian month
+    def merch_to_julian(merch_month)
+      if merch_month > 12 || merch_month <= 0
+        raise ArgumentError
+      end
+
+      if merch_month <= 5
+        merch_month + 7
+      else
+        merch_month - 5
+      end
+    end
+
+    # Converts a julian month to a fiscal month
+    #
+    # @param julian_month [Fixnum] the julian month to convert
+    # @return [Fixnum] the merch month
+    def julian_to_merch(julian_month)
+      if julian_month > 12 || julian_month <= 0
+        raise ArgumentError
+      end
+
+      if julian_month <= 7
+        julian_month + 5
+      else
+        julian_month - 7
+      end
+    end
+    
+    # Given beginning and end dates it will return an array of Fiscal Month's Start date
+    #
+    # @param start_date [Date] the starting date
+    # @param end_date [Date] the ending date
+    # @return [Array] Array of start dates of each Fiscal Month from given dates
+    def merch_months_in(start_date, end_date) 
+      merch_months_combos = merch_year_and_month_from_dates(start_date, end_date)
```

Comment:
`merch_months_combos` is a collection of merch year and months?

Reply:
yep, it is a collection of merch_year and merch_months, it helps with the `merch_months_in` in finding the `start_of_month` of each merch_month inside of the `merch_months_combos`